Lib/asyncio/events.py

```diff
@@ -305,6 +305,16 @@ def set_default_executor(self, executor):
         """
         raise NotImplementedError
 
+    async def start_tls(self, transport, protocol, sslcontext, *,
+                        server_side=False,
```

Comment:
Add `server_hostname` parameter maybe?
At least `create_connection` has it,  checking hostname match increases security.

Reply:
Yes, `server_hostname` should be added. Or perhaps a bunch of `kwargs` to allow passing arbitrary arguments to `wrap_socket` or `wrap_bio`.

Reply:
Sure, will add it.

> Or perhaps a bunch of kwargs to allow passing arbitrary arguments to wrap_socket or wrap_bio.

We don't do that in `loop.create_connection` or in `loop.create_server`, we explicitly pass params there. Most of the stuff can be configured through the `SSLContext` anyways.

Reply:
Yeah... I guess that could be deferred to another PR. However, the `ssl` module may add some other parameters along the way, and it would be nicer to inherit them automatically. For example, Python 3.6 got the `session` parameter: https://docs.python.org/3/library/ssl.html#ssl.SSLContext.wrap_socket

Reply:
Sure. Can you make a PR after this one lands?

Reply:
Clarification: asyncio doesn't use `ssl.wrap_socket` since Python 3.5

Reply:
Yes, but it uses `ssl.wrap_bio` which accepts similar arguments.